plone/app/discussion/conversation.py

```diff
@@ -90,11 +90,19 @@ def enabled(self):
     @property
     def total_comments(self):
         public_comments = [
-            x for x in self._comments.values()
+            x for x in self.values()
             if user_nobody.has_permission('View', x)
         ]
         return len(public_comments)
 
+    def another_total_comments(self):
```

Comment:
> I'm still puzzled about @property being the problem here...

@property removes Acquisition wrappers from 'self'. I don't remember why exactly (some implementation detail of Acquisition), but it's a fact of life.

Instead, if you want a method that acts like a property, you can do this:

    from ComputedAttribute import ComputedAttribute  # Part of the ExtensionClass egg
    class Foo(object):
        def _total_comments(self):  # normal method
            ...
        total_comments = ComputedAttribute(_total_comments, 1)  # 2nd param means something like, deal with one level of acquisition wrapping

p.s. I pity all the pour folks who have tried to use @property as a github username :)